jackson/src/test/kotlin/dev/codified/jackson/CodifiedJacksonModuleTest.kt

```diff
@@ -0,0 +1,154 @@
+package dev.codified.jackson
+
+import com.fasterxml.jackson.core.type.TypeReference
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize
+import org.junit.jupiter.api.Test
+import pl.brightinventions.codified.Codified
+import pl.brightinventions.codified.enums.CodifiedEnum
+import pl.miensol.shouldko.shouldEqual
+
+class CodifiedJacksonModuleTest {
+    val objectMapper = ObjectMapper().findAndRegisterModules()
+
+    @Test
+    fun `can serialize and deserialize known value to codified enum`() {
+        // given
+        val input = Colour.Blue
+
+        // when
+        val output = serdeCodifiedEnum(input)
+
+        // then
+        output.knownOrNull().shouldEqual(input)
+    }
+
+    @Test
+    fun `can serialize and deserialize not known value to codified enum`() {
+        // given
+        val input = "pink"
+
+        // when
+        val output = serdeCodifiedEnum(input)
+
+        // then
+        output.knownOrNull().shouldEqual(null)
+        output.code().shouldEqual("pink")
+    }
+
+    @Test
+    fun `can serialize and deserialize known value to enum property`() {
+        // given
+        val input = HasColour(Colour.Blue)
+
+        // when
+        val output = serde(input)
+
+        // then
+        output.shouldEqual(input)
+    }
+
+    @Test
+    fun `can serialize and deserialize not known value to enum property`() {
+        // given
+        val input = """{"colour": "pink"}"""
+        // when
+        val output = objectMapper.readValue(input, HasColour::class.java)
+        // then
+        output.colour.shouldEqual(null)
+    }
+
```

Comment:
An additional test, complementary to the one above:

```suggestion

    @Test
    fun `can deserialize known value to enum property`() {
        // given
        val input = """{"colour": "blue"}"""
        // when
        val output = objectMapper.readValue(input, HasColour::class.java)
        // then
        output.colour.shouldEqual(Colour.Blue)
    }
```